src/functionaltests/java/com/ericsson/ei/subscriptions/repeatHandler/SubscriptionRepeatHandlerSteps.java

```diff
@@ -0,0 +1,166 @@
+        subscriptionStrWithOneMatch = FileUtils.readFileToString(new File(REPEAT_FLAG_SUBSCRIPTION_COLLECTIONS_WITH_ONE_MATCH), "UTF-8");
+        subscriptionStrWithTwoMatch = FileUtils.readFileToString(new File(REPEAT_FLAG_SUBSCRIPTION_COLLECTIONS_WITH_TWO_MATCH), "UTF-8");
+        aggregatedObject = FileUtils.readFileToString(new File(AGGREGATED_OBJECT_FILE_PATH), "UTF-8");
+        subscriptionWithOneMatch = new JSONObject(subscriptionStrWithOneMatch);
+        subscriptionWithTwoMatch = new JSONObject(subscriptionStrWithTwoMatch);
+    }
+
+    @Given("^Publish events on Message Bus$")
+    public void publish_events_on_Message_Bus() throws IOException, InterruptedException {
+        rulesHandler.setRulePath(RULES_FILE_PATH);
+        sendEiffelEvents(EVENTS_FILE_PATH);
+        List<String> arguments = new ArrayList<>();
+        arguments.add("ongoing=true");
+        List<String> missingArguments = verifyAggregatedObjectInDB(arguments);
+        assertEquals("The following arguments are missing in the Aggregated Object in mongoDB: "
+                + missingArguments.toString(), 0, missingArguments.size());
+    }
+
+    @When("^In MongoDb RepeatFlagHandler collection the subscription has matched the AggrObjectId$")
+    public void in_MongoDb_RepeatFlagHandler_and_subscription_collections_the_subscription_has_matched_the_AggrObjectId() throws IOException {
+        processSubscription(subscriptionStrWithOneMatch, subscriptionWithOneMatch);
+        List<String> resultRepeatFlagHandler = mongoDBHandler.find(dataBaseName, repeatFlagHandlerCollection, subscriptionIdMatchedAggrIdObjQuery);
+        assertEquals(1, resultRepeatFlagHandler.size());
+        assertEquals("\"" + AGGREGATED_OBJECT_ID + "\"", getAggregatedObjectId(resultRepeatFlagHandler, 0));
+    }
+
+    @Then("^I make a DELETE request with subscription name \"([^\"]*)\" to the subscription REST API \"([^\"]*)\"$")
+    public void i_make_a_DELETE_request_with_subscription_name_to_the_subscription_REST_API(String name, String subscriptionEndPoint) throws Exception {
+        MvcResult result = mockMvc.perform(MockMvcRequestBuilders.delete(subscriptionEndPoint + name).accept(MediaType.APPLICATION_JSON)).andReturn();
+        assertEquals(HttpStatus.OK.value(), result.getResponse().getStatus());
+    }
+
+    @Then("^Check in MongoDB RepeatFlagHandler collection that the subscription has been removed$")
+    public void check_in_MongoDB_RepeatFlagHandler_collections_that_the_subscription_has_been_removed() throws IOException {
+        List<String> resultRepeatFlagHandler = mongoDBHandler.find(dataBaseName, repeatFlagHandlerCollection, subscriptionIdMatchedAggrIdObjQuery);
+        assertEquals("[]", resultRepeatFlagHandler.toString());
+        assertTrue(mongoDBHandler.dropDocument(dataBaseName, collectionName, getJSONFromFile(AGGREGATED_OBJECT_FILE_PATH).toString()));
```

Comment:
Why do you need to dropDocument based on AggredgatedObjectFile.... ?

Reply:
because inserting invoking two times if don't drop collection we get double key in db